src/main/java/com/powsybl/openloadflow/network/impl/AbstractLfBus.java

```diff
@@ -111,106 +103,46 @@ public double getTargetQ() {
-        return Optional.ofNullable(controlledBus);
-    }
-
-    public void setControlledBus(AbstractLfBus controlledBus) {
-        Objects.requireNonNull(controlledBus);
-        // check that remote control bus is still the same
-        if (this.controlledBus != null && this.controlledBus.getNum() != controlledBus.getNum()) {
-            throw new PowsyblException("Generators " + getGeneratorIds()
-                    + " connected to bus '" + getId() + "' must control the voltage of the same bus");
-        }
-
-        if (voltageControl) {
-            // check that targetV has a plausible value (wrong nominal voltage issue)
-            double targetVPu = targetV / controlledBus.getNominalV();
-            if (targetVPu < PlausibleValues.MIN_TARGET_VOLTAGE_PU || targetVPu > PlausibleValues.MAX_TARGET_VOLTAGE_PU) {
-                throw new PowsyblException("Controller bus '" + getId() + "' has an inconsistent remote target voltage: "
-                        + targetVPu + " pu");
-            }
-
-            // check target voltage consistency between local and remote control
-            if (controlledBus.hasVoltageControl()) { // controlled bus has also local voltage control
-                double localTargetV = controlledBus.getTargetV() * controlledBus.getNominalV();
-                if (FastMath.abs(this.targetV - localTargetV) > TARGET_V_EPSILON) {
-                    throw new PowsyblException("Bus '" + controlledBus.getId()
-                            + "' controlled by bus '" + getId() + "' has also a local voltage control with a different value: "
-                            + localTargetV + " and " + this.targetV);
-                }
-            }
-
-            // check that if target voltage is consistent with other already existing controller buses
-            List<LfBus> otherControllerBuses = controlledBus.getControllerBuses();
-            if (!otherControllerBuses.isEmpty()) {
-                // just need to check first bus that control voltage
-                otherControllerBuses.stream()
-                        .filter(LfBus::hasVoltageControl)
-                        .findFirst()
-                        .ifPresent(otherControllerBus -> {
-                            double otherTargetV = otherControllerBus.getTargetV() * controlledBus.getNominalV();
-                            if (FastMath.abs(otherTargetV - this.targetV) > TARGET_V_EPSILON) {
-                                LOGGER.error("Bus '{}' control voltage of bus '{}' which is already controlled by at least the bus '{}' with a different target voltage: {} (kept) and {} (ignored)",
-                                        getId(), controlledBus.getId(), otherControllerBus.getId(), otherTargetV, this.targetV);
-                                this.targetV = otherTargetV;
-                            }
-                        });
-            }
+    public void setVoltageControl(boolean voltageControl) {
```

Comment:
You have to renamed ths function for sure. I thought that you can make it using the Mode OFF of the voltageControl ?

Reply:
No clear to me too what is the difference between mode of the VoltageControl and property voltageControlEnabled ? Is mode really usefull as mode is kind of result of individual voltage control enable status of all controllers?

Reply:
I removed the mode, as `VoltageControl` is only structural, the state (`voltageControlEnabled`) remains in `LfBus`.

Reply:
`setVoltageControl()` is moreover renamed into `setVoltageControlEnabled()`